# check fitting 

## fitting is finished or not

In [23]:
dir1='/Users/astro/Documents/notebooks/zs/stack/3arcsec/fitting/'
import glob
import numpy as np

f_ppxfzoom_v2=glob.glob(dir1+'*-ppxf-zoom_v2.jpg')
f_ppxf=glob.glob(dir1+'*-ppxf.jpg')
f_con_goodpixel=glob.glob(dir1+'*_con_goodpixel.fits')
f_con_v2=glob.glob(dir1+'*_con_v2.fits')
f_emission=glob.glob(dir1+'*_emission.jpg')
f_show=glob.glob(dir1+'*_show.jpg')
f_snr_em=glob.glob(dir1+'*_snr_em.jpg')
f_txt_all=glob.glob(dir1+'*txt')

print('ppxf_zoom_v2.jpg : ',len(f_ppxfzoom_v2))
print('ppxf.jpg : ',len(f_ppxf))
print('con_goodpixel : ',len(f_con_goodpixel))
print('con_v2 : ',len(f_con_v2))
print('emission : ',len(f_emission))
print('show : ',len(f_show))
print('snr_em : ',len(f_snr_em))
print('txt_all : ',len(f_txt_all))

p_ppxfzoom_v2=[n[len(dir1):-len('-ppxf-zoom_v2.jpg')] for n in f_ppxfzoom_v2]
p_ppxf=[n[len(dir1):-len('-ppxf.jpg')] for n in f_ppxf]
p_con_goodpixel=[n[len(dir1):-len('_con_goodpixel.fits')] for n in f_con_goodpixel]
p_con_v2=[n[len(dir1):-len('_con_v2.fits')] for n in f_con_v2]
p_emission=[n[len(dir1):-len('_emission.jpg')] for n in f_emission]
p_show=[n[len(dir1):-len('_show.jpg')] for n in f_show]
p_snr_em=[n[len(dir1):-len('_snr_em.jpg')] for n in f_snr_em]
p_txt_all=[n[len(dir1):-len('txt')] for n in f_txt_all]


    

ppxf_zoom_v2.jpg :  6467
ppxf.jpg :  6467
con_goodpixel :  6467
con_v2 :  6467
emission :  5861
show :  5861
snr_em :  6467
txt_all :  6470


最终是拟合了6467个光谱、 但是其实有6500个stack之后的数据，其中32个是红移数据<0;


其中有1个出现了Assertion error ： 8239-3701 


In [24]:
p_ppxf=np.array(p_ppxf)
# np.equal(np.array(p_ppxfzoom_v2),np.array(p_ppxf))
p_ppxfzoom_v2=np.array(p_ppxfzoom_v2)
np.array_equal(p_ppxf,p_ppxfzoom_v2)
# p_ppxfzoom_v2

True

In [25]:
np.array_equal(p_ppxf,np.array(p_con_goodpixel))

True

In [26]:
np.array_equal(np.array(p_con_goodpixel),np.array(p_con_v2))

True

In [27]:
np.array_equal(np.array(p_con_v2),np.array(p_snr_em))

True

In [28]:
np.array_equal(np.array(p_emission),np.array(p_show))

True